sentinel2preprocessing/l1c_to_l2a_conversion.py

```diff
@@ -0,0 +1,63 @@
+import subprocess
+import os
+import logging
+import time
+from typing import List
+
+logger = logging.getLogger(__name__)
+logging.basicConfig()
+
+
+class Sentinel2Converter:
+    """
+    Class for converting Sentinel2 L1C to L2A images
+    """
+
+    def __init__(self, verbose):
+        """
+        :param verbose: bool, flag, print logging information, default: False
+        """
+        if verbose:
+            logger.setLevel(logging.INFO)
+        else:
+            logger.setLevel(logging.CRITICAL)
+
+    def convert_l1c_to_l2a(self, input_tile_path, output_dir_path) -> bool:
+        if not os.path.exists(input_tile_path):
+            logger.error(f"Check that your input tile directory exists: {input_tile_path}")
+            return False
+        # Creating these folders is required for correct sen2cor processing
+        os.makedirs(os.path.join(input_tile_path, "AUX_DATA"), exist_ok=True)
+        os.makedirs(os.path.join(input_tile_path, "HTML"), exist_ok=True)
+        logger.info(f"Started converting {input_tile_path}")
+        process = subprocess.run(['L2A_Process', f'--output_dir={output_dir_path}', f'{input_tile_path}'],
```

Comment:
'L2A_Process' - make path to the installed sen2cor as variable with default value='L2A_Process'

Reply:
I'm not really sure what you want me to do here. L2A_Process is one of the scripts, that is part of sen2cor. When installing I add sen2cor/bin to PATH environment variable. Do you want me to do it in a different manner?

Reply:
The idea is just to add path to executable script 'L2A_Process' as var that could be changed by user with default value 'L2A_Process' if user has used our script for installation. But also it can install sen2cor not in home dir.

Reply:
Ok, added a new parameter to the function in case we want to change where L2A_Process is stored